cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -108,26 +112,55 @@ func (t CertificateTree) CreateTree(ic *kubeadmapi.InitConfiguration) error {
 			return err
 		}
 
-		caCert, caKey, err := NewCACertAndKey(cfg)
-		if err != nil {
-			return err
+		var caKey *rsa.PrivateKey
+
+		caCert, err := pkiutil.TryLoadCertFromDisk(ic.CertificatesDir, ca.BaseName)
+		if err == nil {
+			// Cert exists already, make sure it's valid
+			if !caCert.IsCA {
+				return fmt.Errorf("certificate %q should be a CA but is not", ca.Name)
```

Comment:
`certificate %q should be a CA but is not`
-> 
`certificate %q is not a CA`
?

Reply:
I'm ok with the statement.